plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakePageSourceProvider.java

```diff
@@ -255,4 +279,21 @@ public static Optional<HiveColumnHandle> toHiveColumnHandle(DeltaLakeColumnHandl
                 throw new IllegalArgumentException("Unsupported column mapping: " + columnMapping);
         }
     }
+
+    private static ConnectorPageSource createPages(long totalRowCount)
+    {
+        // This implementation of the page source is used whenever a query doesn't reference any data
+        // columns. We need to limit the number of rows per page in case there are projections
+        // in the query that can cause page sizes to explode. For example: SELECT rand() FROM some_table
+        int maxPageSize = 100_000;
```

Comment:
Can we use the same value as `PageProcessor#MAX_BATCH_SIZE` ?

Reply:
I based this on `io.trino.plugin.elasticsearch.CountQueryPageSource#BATCH_SIZE` (because i had a context there -- it was added per my request), but I used a bigger value, as 10K in Data lake context felt quite small

I can use `PageProcessor#MAX_BATCH_SIZE`, but it lacks a comment explaining context & purpose. And also is even smaller.

Ideally, I would prefer to use MAX_INT here, and let the engine worry about perf penalty and split pages IF needed, but it would require your help @raunaqmorarka to get done. 

Thoughts?

Reply:
Orc and parquet readers self regulate the page sizes they produce (both on row count and data size). Once a connector produces a big page, it uses up memory. So rather than having that increased memory consumption and then extra work to figure out if splitting is needed and then split pages, I would prefer to rely on connectors to avoid producing large pages themselves.
In this particular case, the only way the connector would produce a page with large size is if there are synthetic columns which are not Rle blocks (maybe row_id). I suggested 8K only for consistency with orc and parquet reader max batch size (e.g. OrcReader#MAX_BATCH_SIZE, ParquetReaderOptions#DEFAULT_MAX_READ_BLOCK_ROW_COUNT).

Reply:
i used DEFAULT_MAX_PAGE_SIZE_IN_BYTES to calculate page size accomodating for additinal e.g. bigint block to get projected _after_ the page leaves the connector

(when row_id is not needed, we could produce pages arbitrarily big, as they would use O(1) memory, but i am sticking to the more conservative approach)